esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -2262,7 +2267,9 @@ int EsdlBindingImpl::onGetInstantQuery(IEspContext &context,
                     StringBuffer ns, schemaLocation;
                     generateNamespace(context, request, srvdef->queryName(), mthdef->queryName(), ns);
 
-                    getSchemaLocation(context, request, schemaLocation);
+                    if(m_pESDLService->m_returnSchemaLocation)
+                        getSchemaLocation(context, request, ns, schemaLocation);
```

Comment:
Unless there's a use case where we need the schema location regardless of the configuration value, I would prefer `getSchemaLocation` to include the configuration check instead of expecting every use of the method to wrap the call with a check.

I suppose this depends on the answer to Tony's question about soap faults unconditionally including the schema.

Reply:
I wasn't sure if we'd really need/want the flexibility to have three states: omit from responses only/always return/omit from responses and faults. I opted to keep the behavior consistent with scapps in that the soap fault always returns the schemaLocation, but if you all think there should be even more flexibility it could be changed.

Reply:
This a big improvement.  Any more flexibility would be nice but a low priority.  We can always improve later.